semantics/language/translation/configuration.k

```diff
@@ -93,25 +86,31 @@ module C-CONFIGURATION
                </goto-loop-stack>
           </goto-calc>
      </goto-map-calc>
-     <control>
-          <curr-tu color="lightgray"> "" </curr-tu>
-          <curr-function color="lightgray">
-               file-scope
-          </curr-function>
-          <curr-program-loc color="black">
-               UnknownCabsLoc
-          </curr-program-loc>
-
-          // Toggle to allow writes and other special treatment for
-          // initializers.
-          <initializing> 0 </initializing>
-
-          <control-details> // TODO(chathhorn): dumb name
+     <elab-calc>
+          <elaborator multiplicity="*">
```

Comment:
Performance is less of an issue in the translation semantics but still
significant insofar as it can take half an hour to an hour in the current
rewrite engine to compile a large real code project. But we can leave it
this way for now and run experiments with changing it if things seem so
slow.

On Thu, Jun 4, 2015 at 4:16 PM, Andrei Stefanescu <notifications@github.com>
wrote:

> In semantics/language/translation/configuration.k
> <https://github.com/kframework/c-semantics/pull/105#discussion_r31766593>:
>
> > -     <control>
> > -          <curr-tu color="lightgray"> "" </curr-tu>
> > -          <curr-function color="lightgray">
> > -               file-scope
> > -          </curr-function>
> > -          <curr-program-loc color="black">
> > -               UnknownCabsLoc
> > -          </curr-program-loc>
> > -
> > -          // Toggle to allow writes and other special treatment for
> > -          // initializers.
> > -          <initializing> 0 </initializing>
> > -
> > -          <control-details> // TODO(chathhorn): dumb name
> > +     <elab-calc>
> > +          <elaborator multiplicity="*">
>
> This is only in the translation semantics, so performance is less of an
> issue.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kframework/c-semantics/pull/105/files#r31766593>.
>